drivers/wifi/simplelink/simplelink_sockets.c

```diff
@@ -537,30 +553,40 @@ static int simplelink_poll(struct pollfd *fds, int nfds, int msecs)
 
 	for (i = 0; i < nfds; i++) {
 		fds[i].revents = 0;
-		fd = fds[i].fd;
-		if (fd < 0) {
+		if (fds[i].fd < 0) {
 			continue;
+		} else if ((fds[i].fd >= ARRAY_SIZE(simplelink_offload_fd)) ||
+			   (simplelink_offload_fd[fds[i].fd] == -1)) {
```

Comment:
For example, here'd you use something like z_get_fd_obj(fds[i].fd, &simplelink_socket_fd_op_vtable, ENOTSUPP).